pkg/logging/types.go

```diff
@@ -0,0 +1,75 @@
+/*******************************************************************************
+ * Copyright 2023 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+package logging
+
+import (
+	"log/slog"
+)
+
+type LogLevel string
+type LogKey string
+
+const (
+	TraceLevel LogLevel = "trace"
+	InfoLevel  LogLevel = "info"
+	DebugLevel LogLevel = "debug"
+	WarnLevel  LogLevel = "warn"
+	ErrorLevel LogLevel = "error"
+
+	LevelTrace slog.Level = slog.Level(-8)
+
+	CorrelationKey LogKey = "correlation-id"
+
+	lineNumberAttrKey  = "line-number"
+	applicationAttrKey = "application"
+	hostnameAttrKey    = "hostname"
+)
+
+type LoggingInfo struct {
+	MinLogLevel LogLevel `json:"minLogLevel,omitempty"`
+}
+
```

Comment:
@o-eissa We can likely dispense with the following two functions. We previously had a `LogLevel` enum in our custom provider to indicate the status of log messages. `slog` has a similar enum with equivalent values except for Trace. I suggest we eliminate the legacy provider enum altogether and simply use the `slog` enum directly. No need for the mapping. If at some point we determine some finer grained logging beyond `Debug` is necessary, we'll cross that bridge when we come to it.

When converting the `slog.Level` enum values to a string, you can likely simply use `string(slog.LevelInfo)` without need for a separate function that returns literal values.

Reply:
Changes done, found out ```string(slog.LevelInfo)``` returns string value of the underlying int of the level, but ```slog.LevelInfo.String()```does the job.
Also in the future, whenever any custom level needed, we can simply add a new const as follows:
 ```LevelTrace slog.Level = slog.Level(-8)```